### ISSUE TITLE
Setting C++ standard version from outside

### ISSUE THREAD

**Issue by NikolausDemmel** [stevenlovegrove/Pangolin#469]
I build Pangolin as an external dependency and would like to the standard version to 17 without modifying the source. Currently, it seems that the line

```cmake
set (CMAKE_CXX_STANDARD 14)
```

always overwrites what I set from the command line. If I change it to something like

```cmake
if(NOT CMAKE_CXX_STANDARD)
    set (CMAKE_CXX_STANDARD 14)
endif()
```

it seems to do the trick but still default to `14` if `CMAKE_CXX_STANDARD` is not set.

Can this change be included (I can send a PR)? Or is the another / a better way to set the C++ standard version from outside?

**Comment by NikolausDemmel** [comment]
[Looks like](https://cliutils.gitlab.io/modern-cmake/chapters/features/cpp11.html) the proper modern cmake way is something like

```
target_compile_features(... PUBLIC cxx_std_11)
```

which according to a quick test can be overwritten by setting `CMAKE_CXX_STANDARD`. Unfortunately that is a cmake 3.8+ feature.

**Comment by stevenlovegrove** [comment]
I have a new branch (https://github.com/stevenlovegrove/Pangolin/tree/component_refactor) where I have refactored Pangolin into components and tried to use modern cmake. This branch already uses the method you suggested and has a CMake 3.8 dependency. I'd be happy to hear your feedback on how such a refactor would impact your use case.

**Comment by NikolausDemmel** [comment]
Cool. I gave it a quick try, and it seems to solve the issue nicely, but for other reasons it is not quite ready to be used (I guess it's expected for a WIP branch). Note that I tried the "build as external library and find-package from the build directory" use case and not the "add_subdirectory" use case.

Limitations I found:
 - build fails with C++17 (at least on macOS Sierra with XCode 9.2) 
 - the vars to disable building examples / tools / tests / python are missing (`-DBUILD_EXAMPLES=OFF`, `-DBUILD_TESTS=OFF`, `-DBUILD_PANGOLIN_PYTHON=OFF`, ...). Those are not super essential, but would just allow to speed up the build if pangolin is built as an external module as part of your build.
 - I believe eigen is not properly included by the submodules, and thus some of the eigen versions of the APIs that I'm using are missing, in particular in the gl drawing functions (this needs to respect passing a custom `EIGEN_INCLUDE_DIR`)
 - the cmake flag to not export the build directory to `~/.cmake/packages/Pangolin/` is missing (`-DEXPORT_Pangolin=OFF`)
- Pangolin_INCLUDE_DIR / Pangolin_INCLUDE_DIRS are not set after find-package (for legacy usage)
- components are great, but a unifying target to "give me all of pangolin" would be nice as well
